### PR TITLE
feat(commerce): support category field and field suggestions

### PR DESCRIPTION
**Pull request by Spuffynism**
Create a field suggestions generator to power field suggestions and category field suggestions controllers.

[CAPI-868]

[CAPI-868]: https://coveord.atlassian.net/browse/CAPI-868?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/features/commerce/facets/field-suggestions-order/field-suggestions-order-slice.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {fetchQuerySuggestions} from '../../query-suggest/query-suggest-actions';
+import {getFieldSuggestionsOrderInitialState} from './field-suggestions-order-state';
+
+export const fieldSuggestionsOrderReducer = createReducer(
+  getFieldSuggestionsOrderInitialState(),
+  (builder) => {
+    builder.addCase(fetchQuerySuggestions.fulfilled, (_, action) => {
+      return action.payload.fieldSuggestionsFacets ?? [];
```
Comment (BenKandelaars) [inline packages/headless/src/features/commerce/facets/field-suggestions-order/field-suggestions-order-slice.ts:9]: No validation?

Comment (Spuffynism) [reply packages/headless/src/features/commerce/facets/field-suggestions-order/field-suggestions-order-slice.ts:9]: What do you mean? The `fieldSuggestionsFacets` are what we want as state :eyes:

**Review by BenKandelaars — APPROVED** [review]
Looks good

**Review by BenKandelaars — APPROVED** [review]

File: packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts [hunk 2]
```diff
@@ -143,6 +143,99 @@ export function handleCommerceFacetSearchFulfilled<
...
+  if (!payload.fieldSuggestionsFacets) {
+    return;
+  }
+
+  payload.fieldSuggestionsFacets
+    .filter((fieldSuggestionFacet) => !(fieldSuggestionFacet.facetId in state))
+    .filter(
+      (fieldSuggestionFacet) => fieldSuggestionFacet.type === 'hierarchical'
+    )
+    .forEach((fieldSuggestionFacet) => {
```
Comment (louis-bompart) [inline packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts:225]: yike, that's 3 iterations in the array here.
use a single for..of with continue statement instead

File: packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts [hunk 3]
```diff
@@ -143,6 +143,99 @@ export function handleCommerceFacetSearchFulfilled<
...
+  buildEmptyResponse: () => T
+) {
+  if (!payload.fieldSuggestionsFacets) {
+    return;
+  }
+
+  payload.fieldSuggestionsFacets
+    .filter((fieldSuggestionFacet) => !(fieldSuggestionFacet.facetId in state))
+    .filter((fieldSuggestionFacet) => fieldSuggestionFacet.type === 'regular')
+    .forEach((fieldSuggestionFacet) => {
```
Comment (louis-bompart) [inline packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts:191]: yike, that's 3 iterations in the array here.
use a single for..of with continue statement instead

**Review by louis-bompart — APPROVED** [review]
Code lgtm, please do consider the nitpik I mentioned tho

File: packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts (hunk 2 above)
Comment (Spuffynism) [reply packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts]: Done in https://github.com/coveo/ui-kit/pull/3939/commits/5c3d8fd53ed4b79eaf25f9692ac6c62b802242f2

File: packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts (hunk 3 above)
Comment (Spuffynism) [reply packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts]: Done in https://github.com/coveo/ui-kit/pull/3939/commits/5c3d8fd53ed4b79eaf25f9692ac6c62b802242f2

**Pull request merged**